### PR TITLE
Cleanup BlockStore

### PR DESCRIPTION
**Pull request by rongrong**
### What changes are proposed in this pull request?

Removing some unnecessary APIs in BlockStore.

### CODE REVIEW

**Review by beinan — COMMENTED** [review]
Great, thank you for making our code better!  Just a couple comments, thx!

File: core/server/worker/src/main/java/alluxio/worker/block/BlockMetadataManager.java [hunk 1]
```diff
@@ -324,29 +324,14 @@ public StorageDir getDir(BlockStoreLocation location) {
    * @throws BlockDoesNotExistException when block id can not be found
    */
   public TempBlockMeta getTempBlockMeta(long blockId) throws BlockDoesNotExistException {
```
[the remark this answers is not in the document]
Comment (rongrong) [reply core/server/worker/src/main/java/alluxio/worker/block/BlockMetadataManager.java:326]: We could but that's out of the scope if this PR. We can do it in the future.

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java [hunk 2]
```diff
@@ -641,13 +625,15 @@ public void closeUfsBlock(long sessionId, long blockId)
       throws BlockAlreadyExistsException, IOException, WorkerOutOfSpaceException {
     try {
       mUnderFileSystemBlockStore.closeReaderOrWriter(sessionId, blockId);
-      if (mLocalBlockStore.getTempBlockMeta(sessionId, blockId) != null) {
+      if (mLocalBlockStore.hasTempBlockMeta(blockId)) {
         try {
           commitBlock(sessionId, blockId, false);
-        } catch (BlockDoesNotExistException e) {
+        }
+        catch (BlockDoesNotExistException e) {
```
[the remark this answers is not in the document]
Comment (rongrong) [reply core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java:632]: `commitBlockLocked` throws these exceptions.

**Review by beinan — APPROVED** [review]

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Pull request merged**